api/controllers/ApplicationTypeController.js

```diff
@@ -314,51 +314,50 @@ module.exports = {
-                                                }).then(function() {
-                                                    const appTypeRedirect = {
-                                                        2: '/business-document-quantity?pk_campaign=Premium-Service&pk_kwd=Premium',
-                                                        3: '/business-document-quantity?pk_campaign=DropOff-Service&pk_kwd=DropOff',
-                                                        4: '/eligibility/apostille-accepted-in-destination',
-                                                    };
-
-                                                    const redirectUrl =
-                                                        appTypeRedirect[
-                                                            req.session.appType
-                                                        ];
-                                                    if (redirectUrl) {
-                                                        return res.redirect(
-                                                            redirectUrl
-                                                        );
-                                                    }
-                                                    sails.log.error('serviceType number not found');
-                                                    return res.serverError();
-                                                });
-                                            });
-                                        });
-                                    }
-                                    else  if (req.session.appType == 1){
+
+                                    if (req.session.appType === 1) {
                                         req.session.summary = false; // Reset summary session variable
                                         req.session.useDetails = true;
                                         HelperService.resetAddressSessionVariables(req,res);
 
                                         return res.redirect('/choose-documents-or-skip?pk_campaign=Standard-Service&pk_kwd=Standard');
                                     }
+
+                                    if (req.session.appType === 4) {
+                                        return res.redirect('/eligibility/apostille-accepted-in-destination');
+                                    }
+
+                                    const UserModels = getUserModels(
+                                        req._sails.config
+                                            .userServiceSequelize
+                                    );
+                                    const userDataFromDB = await UserModels.User.findOne({where:{email:req.session.email}});
+                                    const accountDetailsFromDB = await UserModels.AccountDetails.findOne({where:{user_id:userDataFromDB.id}});
+
+                                    await UsersBasicDetails.create({
```

Comment:
Hey Ali, I'm not quire sure I understand the question. 

1. Is it more to do with positioning? The create could run before the two variables above it others and could cause a race condition since it needs information from them?

2. Or more for necessity. It the create needed at all and can it be removed?

---

If your question is the first. To my understanding then the JS code is read line by line, the await conditions will need to complete before the code moves on to the next line. So in theory, there shouldn't be a race condition. But if for some reason there is, it should be caught by the catch function on line **362**.

For the second question. I'd admit it is a bit weird to create a new user when the user is creating the application especially if the user already has details saved in the DB. I had a chat with Kainos about this and that's how they received the service. I think for a standard paper application, this makes sense because you can create one of these without an account. But for the other types of applications, e-App included, you need an account.

It's not very easy to follow but on line **326** eApps have an appType of 4, so they wouldn't hit any of this code, a user is created later on in the flow for eApps.

Reply:
Ah my bad.  The await should cover it as the function calls will effectively get called synchronously.